src/sentry/static/sentry/app/views/performance/traceDetails/transactionBar.tsx

```diff
@@ -0,0 +1,270 @@
+import React from 'react';
+
+import Count from 'app/components/count';
+import * as DividerHandlerManager from 'app/components/events/interfaces/spans/dividerHandlerManager';
+import {TraceFull} from 'app/utils/performance/quickTrace/types';
+
+import {
+  ConnectorBar,
+  DividerLine,
+  DividerLineGhostContainer,
+  StyledIconChevron,
+  TRANSACTION_ROW_HEIGHT,
+  TransactionBarTitle,
+  TransactionBarTitleContainer,
+  TransactionRow,
+  TransactionRowCell,
+  TransactionRowCellContainer,
+  TransactionTreeConnector,
+  TransactionTreeToggle,
+  TransactionTreeToggleContainer,
+} from './styles';
+import {toPercent} from './utils';
+
+const TOGGLE_BUTTON_MARGIN_RIGHT = 16;
+const TOGGLE_BUTTON_MAX_WIDTH = 30;
+export const TOGGLE_BORDER_BOX = TOGGLE_BUTTON_MAX_WIDTH + TOGGLE_BUTTON_MARGIN_RIGHT;
+const MARGIN_LEFT = 0;
+
+type Props = {
+  transaction: TraceFull;
+  isLast: boolean;
+  continuingDepths: Array<number>;
+  isExpanded: boolean;
+  toggleExpandedState: () => void;
+};
+
+function getOffset(generation) {
+  return generation * (TOGGLE_BORDER_BOX / 2) + MARGIN_LEFT;
+}
+
+class TransactionBar extends React.Component<Props> {
```

Comment:
What's the plan long term for this? Is this just temporary while building trace details and we'll merge some of this with the existing span bar, or do we need two dedicated tree-panel UX's?

Reply:
Ideally, we would consolidate them into a single component with a unified type that we pass to generate them.

Though for the time being, I want to get this in first. There are some differences between the two, especially in the interactions (i.e. `SpanBar` has an `onClick` that expands the details, while we aren't sure what that means for `TransactionBar` yet) that we want to look at more closely before I go down the path of merging the two.

Reply:
Also, something to keep in mind is that we also have that transaction comparison view that likely shares some similarities as well.